tests/components/crownstone/test_config_flow.py

```diff
@@ -0,0 +1,531 @@
+    assert result["step_id"] == "usb_sphere_config"
+    assert serial_mock.call_count == 1
+
+    # select a sphere
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={CONF_USB_SPHERE: "sphere_name_1"}
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result["data"] == entry_data_with_usb
+    assert result["options"] == entry_options_with_usb
+
+
+@patch(
+    "serial.tools.list_ports.comports", MagicMock(return_value=[get_mocked_com_port()])
+)
+async def test_successful_login_with_manual_usb_path(hass: HomeAssistant):
+    """Test flow with correct login and usb configuration."""
+    entry_data_with_manual_usb = create_mocked_entry_data_conf(
+        email="example@homeassistant.com",
+        password="homeassistantisawesome",
+    )
+    entry_options_with_manual_usb = create_mocked_entry_options_conf(
+        usb_path="/dev/crownstone-usb",
+        usb_sphere="sphere_id_0",
+    )
+
+    result = await start_config_flow(
+        hass, get_mocked_crownstone_cloud(create_mocked_spheres(1))
+    )
+    # should show usb form
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "usb_config"
+
+    # select manual from the list
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={CONF_USB_PATH: MANUAL_PATH}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "usb_manual_config"
+
+    # enter USB path
+    path = "/dev/crownstone-usb"
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={CONF_USB_MANUAL_PATH: path}
+    )
+
+    # since we only have 1 sphere here, test that it's automatically selected and
+    # creating entry without asking for user input
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result["data"] == entry_data_with_manual_usb
+    assert result["options"] == entry_options_with_manual_usb
+
+
+@patch(
+    "serial.tools.list_ports.comports", MagicMock(return_value=[get_mocked_com_port()])
+)
+@patch(
+    "homeassistant.components.usb.get_serial_by_id",
+    return_value="/dev/serial/by-id/crownstone-usb",
+)
+async def test_options_flow_setup_usb(serial_mock: MagicMock, hass: HomeAssistant):
+    """Test options flow init."""
+    configured_entry_data = create_mocked_entry_data_conf(
+        email="example@homeassistant.com",
+        password="homeassistantisawesome",
+    )
+    configured_entry_options = create_mocked_entry_options_conf(
+        usb_path=None,
+        usb_sphere=None,
+    )
+
+    # create mocked entry
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=configured_entry_data,
+        options=configured_entry_options,
+        unique_id="account_id",
+    )
+    entry.add_to_hass(hass)
+
+    result = await start_options_flow(
```

Comment:
Please set up the integration before starting the options flow to make sure the config flow module is loaded.

Reply:
I have a fixture on line 32 which runs the integration setup, with `autouse` set to `True`, i would assume the integration setup has returned True before this code runs?

Reply:
That fixture only patches a function in the init module. We need to set up the integration to make sure the config flow module is loaded.